u_http.go

```diff
@@ -0,0 +1,565 @@
+package tls
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/net/http2"
+	"golang.org/x/net/idna"
+	"golang.org/x/net/proxy"
+)
+
+// uhttpLogger is the type of the optional logger. The log.Default
+// logger in the standard library matches this interface.
+type uhttpLogger interface {
+	Printf(fmt string, v ...interface{})
+	Print(v ...interface{})
+}
+
+// uhttpSilentLogger is the silent logger.
+type uhttpSilentLogger struct{}
+
+// Printf implements uhttpLogger.Printf.
+func (ul *uhttpSilentLogger) Printf(fmt string, v ...interface{}) {}
+
+// Print implements uhttpLogger.Print.
+func (ul *uhttpSilentLogger) Print(v ...interface{}) {}
+
+// uhttpLog is the default logger.
+var uhttpLog uhttpLogger = &uhttpSilentLogger{}
+
+// init checks whether the user wants verbose logs.
+func init() {
+	if os.Getenv("UHTTP_VERBOSE") == "1" {
```

Comment:
Please add a function that will let users enable verbose logging without having to set environment variables (and mention it in the readme).

Reply:
Sure! (The rationale of this design with the environment variable was to do what the standard library does in terms of logging: they check the `GODEBUG` environment variable to determine whether to emit debug events.)